src/commands.rs

```diff
@@ -562,29 +562,41 @@ struct SplitArgs {
 /// Unlike `-s` and `-r`, the `-b` mode takes the destination into account
 /// when calculating the set of revisions to rebase. That set includes the
 /// specified revision and all ancestors that are not also ancestors
-/// of the destination. It also includes all descendants of those commits. For
-/// example, `jj rebase -b B -d D` or `jj rebase -b C -d D`  would transform
-/// your history like this:
+/// of the destination. It also includes all descendants of those commits. `jj
+/// rebase -b X -d Y` thus rebases the commits in the revset `(Y..X):`, making
+/// it equivalent to `jj rebase -s 'roots(Y..X)' -d Y` (except that `-s` doesn't
+/// yet support that case when the `roots()` expression has more than 1
+/// revision). For example, `jj rebase -b C -d F` or `jj rebase -b E -d F`
+/// would transform your history like this (letters followed by an apostrophe
+/// are post-rebase versions):
```

Comment:
Suggestion:
* I find it a bit difficult to parse the plaintext description of the commits being rebased.
* Maybe we should swap the order of `-b` and `-s` if we are using `-s` as an example. 
* The parenthetical on `-s` seems distracting.

Maybe something like...

```
With `-b`, it rebases the whole "branch" containing the specified revision.

A "branch" is the set of commits that includes:

* the specified revision and ancestors that are not also ancestors of the destination
* all descendants of those commits

In other words, `jj rebase -b X -d Y` rebases commits in the revset `(Y..X):` (which is equivalent to `jj rebase -s 'roots(Y..X)' -d Y` for a single root). For example...
```

Or alternatively, we could phrase it in terms of the `roots()` formulation (which is closer to how I think of it). I don't know if that's more inaccurate, but (with some additional examples) that would be something like:

```
When running `jj rebase -B X -d Y`, a "branch" is the set of commits that includes:

* The earliest ancestors of the specified revision (X) that is neither the destination (Y) nor an ancestor of the destination (equivalent to the revset `roots(Y..X)`).
* All descendants of those earliest ancestors (equivalent to the revset `roots(Y..X):`)
```

It's just personal preference to me though. Maybe we should get opinions from fresher eyes.

Reply:
> * Maybe we should swap the order of `-b` and `-s` if we are using `-s` as an example.

I considered that but left `-b` first because it's the default. Maybe that's not a good reason. You're probably right that it's better to start with the simpler `-s`. Done.

> * The parenthetical on `-s` seems distracting.

Removed.

> Maybe something like...

Thanks! I used this version. It's how I think of it.